src/libstd/sys/windows/stdio.rs

```diff
@@ -246,6 +246,11 @@ impl Stdout {
     pub fn new() -> io::Result<Stdout> {
         Ok(Stdout)
     }
+    pub fn should_be_line_buffered(&self) -> bool {
+        // FIXME: Fill in. I don't know how to check whether output is
+        // redirected on Windows.
+        true
```

Comment:
On Windows there's a `GetConsoleMode` function you can call. It's [used in the `atty` library's Windows implementation](https://github.com/softprops/atty/blob/master/src/lib.rs#L51-L81). It looks like that crate also has some more complex heuristics for Windows but those might not be so important here.